app/app/src/main/java/com/example/myapplication/MainActivity.java

```diff
@@ -365,6 +372,30 @@ public void handleMessage(Message msg) {
                             } catch (IOException e) {
                             }
                         }
+
+                        // This part changes the pulse animation on EEG change
+                        if ((msg.arg1 >= 1) && (msg.arg1 <= 10)) {
+                            pulse.setConcentration(1);
+                        } else if ((msg.arg1 >= 11) && (msg.arg1 <= 20)) {
+                            pulse.setConcentration(2);
+                        } else if ((msg.arg1 >= 21) && (msg.arg1 <= 30)) {
+                            pulse.setConcentration(3);
+                        } else if ((msg.arg1 >= 31) && (msg.arg1 <= 40)) {
+                            pulse.setConcentration(4);
+                        } else if ((msg.arg1 >= 41) && (msg.arg1 <= 50)) {
+                            pulse.setConcentration(5);
+                        } else if ((msg.arg1 >= 51) && (msg.arg1 <= 60)) {
+                            pulse.setConcentration(6);
+                        } else if ((msg.arg1 >= 61) && (msg.arg1 <= 70)) {
+                            pulse.setConcentration(7);
+                        } else if ((msg.arg1 >= 71) && (msg.arg1 <= 80)) {
+                            pulse.setConcentration(8);
+                        } else if ((msg.arg1 >= 81) && (msg.arg1 <= 90)) {
+                            pulse.setConcentration(9);
+                        } else if (msg.arg1 > 90) {
+                            pulse.setConcentration(10);
+                        }
```

Comment:
Does anyone know of a "prettier" way to handle above? The previous version was a switch case, but I can't do that with a boolean.

Reply:
Won't this be good enough:

```java
pulse.setConcentration((msg.arg1 / 10) + 1);
```

In any case, you'll probably want to make a special case for `0` just to be explicit about things.

Reply:
Thank you, that's so much cleaner! I've created a new issue for it because this branch was merged before I saw this.